molgenis-data-security/src/main/java/org/molgenis/data/security/permission/PermissionServiceImpl.java

```diff
@@ -0,0 +1,387 @@
+
+  private final MutableAclService mutableAclService;
+  private final PermissionInheritanceResolver inheritanceResolver;
+  private final ObjectIdentityService objectIdentityService;
+  private final DataService dataService;
+  private final MutableAclClassService mutableAclClassService;
+  private final UserRoleTools userRoleTools;
+  private final EntityHelper entityHelper;
+
+  public PermissionServiceImpl(
+      MutableAclService mutableAclService,
+      PermissionInheritanceResolver inheritanceResolver,
+      ObjectIdentityService objectIdentityService,
+      DataService dataService,
+      MutableAclClassService mutableAclClassService,
+      UserRoleTools userRoleTools,
+      EntityHelper entityHelper) {
+    this.mutableAclService = requireNonNull(mutableAclService);
+    this.inheritanceResolver = requireNonNull(inheritanceResolver);
+    this.objectIdentityService = requireNonNull(objectIdentityService);
+    this.dataService = requireNonNull(dataService);
+    this.mutableAclClassService = requireNonNull(mutableAclClassService);
+    this.userRoleTools = requireNonNull(userRoleTools);
+    this.entityHelper = requireNonNull(entityHelper);
+  }
+
+  @Override
+  public Set<LabelledType> getTypes() {
+    Set<LabelledType> types = new HashSet<>();
+    for (String typeId : mutableAclClassService.getAclClassTypes()) {
+      String entityTypeId = entityHelper.getEntityTypeIdFromType(typeId);
+      String label = entityHelper.getLabel(typeId);
+      types.add(LabelledType.create(typeId, entityTypeId, label));
+    }
+    return types;
+  }
+
+  @Override
+  public Set<LabelledObject> getObjects(String typeId, int page, int pageSize) {
+    entityHelper.checkEntityTypeExists(typeId);
+    return objectIdentityService
+        .getObjectIdentities(typeId, pageSize, (page - 1) * pageSize)
+        .stream()
+        .map(this::getLabelledObject)
+        .collect(toSet());
+  }
+
+  private LabelledObject getLabelledObject(ObjectIdentity objectIdentity) {
+    String label =
+        entityHelper.getLabel(objectIdentity.getType(), objectIdentity.getIdentifier().toString());
+    return LabelledObject.create(objectIdentity.getIdentifier().toString(), label);
+  }
+
+  @Override
+  public Set<PermissionSet> getSuitablePermissionsForType(String typeId) {
+    entityHelper.checkEntityTypeExists(typeId);
+    Set<PermissionSet> permissions;
+    switch (typeId) {
+      case EntityTypeIdentity.ENTITY_TYPE:
+      case PackageIdentity.PACKAGE:
+        permissions =
+            Sets.newHashSet(
+                PermissionSet.READMETA,
+                PermissionSet.COUNT,
+                PermissionSet.READ,
+                PermissionSet.WRITE,
+                PermissionSet.WRITEMETA);
+        break;
+      case PLUGIN:
+        permissions = Sets.newHashSet(PermissionSet.READ);
+        break;
+      default: // RLS
+        permissions = Sets.newHashSet(PermissionSet.READ, PermissionSet.WRITE);
```

Comment:
just to be on the safe side check if startswith 'entity-' and otherwise throw an UnknownTypeException?